src/OrleansProviders/Streams/Common/PooledCache/IObjectPool.cs

```diff
@@ -23,6 +24,11 @@ public interface IObjectPool<T>
         /// </summary>
         /// <param name="resource"></param>
         void Free(T resource);
+
+        /// <summary>
+        /// Unique id of current blockId
+        /// </summary>
+        string Id { get; }
```

Comment:
it is one dimension for object pool metrics. but I guess it can just stay in ObjectPoolMonitorDimemsions.